packages/nimble-components/src/wafer-map/specs/features/grid-dimension.md

```diff
@@ -0,0 +1,26 @@
+# Nimble Wafer Map Grid Dimensions
+
+## Problem Statement
+
+The Wafer Map displays dies on a canvas inside a circular border based on an internal grid which is calculated based on the data input.
+This approach is limiting the performance of the wafer and the grid dimensions can be erroneous if the displayed data is just a subset of a larger data set.
+To mitigate these problems, we will reveal three new API inputs which will allow the users to set the grid dimensions themselves, thus saving processing time and correcting inconsistencies generated by the lack of data context.
+
+## Links To Relevant Work Items and Reference Material
+
+-   [Wafer Map Spec](../README.md)
+-   [Interactive Wafer Map Improvements](https://dev.azure.com/ni/DevCentral/_wiki/wikis/Stratus/66917/Interactive-Wafer-Map-Improvements)
+
+## Implementation / Design
+
+The new attributes and properties will be described in the component definition. They will be created with change listeners which will trigger the appropriate updates in the wafer.
+
+The inputs will be validated before use. If they are undefined or the data is missing the previous behavior of calculating the values based on the input data will be implemented.
+
+## Alternative Implementations / Designs
+
+The API can be changed to use four numeric inputs for each of the values, or a new Type can be created to structure all the dimensions together.
+
+## Open Issues
+
+-   Should the detection of partial data on the new inputs trigger a visual warning?
```

Comment:
The strategy we've taken with the table is:
1. if the client developer configured the component incorrectly, the component renders blank to make it very obvious that something is wrong
2. the component [exposes "validity" APIs](https://nimble.ni.dev/storybook/?path=/docs/incubating-table--docs) to report whether the configuration is valid and explain what's wrong with it. These can be called in tests to assert there are no problems or when debugging to see why the table is rendering blank (they wouldn't typically be called in production)

We've been pretty happy with that approach so you could consider it for this component. Item 2 may be "nice to have" but I think defining consistent behavior (and autotesting that behavior) when configuration is invalid is important.

Reply:
I think we can try and implement this approach covering only this case for now and expand it in a later feature or tech debt